### PR TITLE
rest of 5B320

### PR DESCRIPTION
**Pull request by HailSanta**

### CODE REVIEW

**Comment by BowserSlug** [comment]
⚠️ This PR introduces 10 warnings:
- src/5B320.c:1553: warning: initialization from incompatible pointer type
- src/5B320.c:1553: warning: initialization from incompatible pointer type
- src/5B320.c:1553: warning: initialization from incompatible pointer type
- src/5B320.c:1553: warning: initialization from incompatible pointer type
- src/5B320.c:1553: warning: initialization from incompatible pointer type
- src/5B320.c:1553: warning: initialization makes integer from pointer without a cast
- src/5B320.c:1553: warning: initialization makes integer from pointer without a cast
- src/5B320.c:1553: warning: initialization makes integer from pointer without a cast
- src/5B320.c:1553: warning: initialization makes integer from pointer without a cast
- src/5B320.c:1553: warning: initialization makes integer from pointer without a cast

File: src/item_entity_scripts.c [hunk 1]
```diff
@@ -0,0 +1,1875 @@
...
+	ICON_key_Gift_palette,
+	ICON_key_Gift_palette,
+	ICON_key_Gift_palette,
+	ICON_key_Gift_palette,
+	ICON_anim_heart_0_palette,
+	ICON_anim_coin_0_palette,
+	ICON_anim_heart_piece_palette,
+	ICON_anim_sp_0_palette,
+	ICON_anim_hp_0_palette,
+	ICON_anim_fp_0_palette,
```
Comment (ethteck) [inline src/item_entity_scripts.c:1873]: spaces instead of tabs

**Review by ethteck — CHANGES_REQUESTED** [review]
Grood! Please replace tabs with 4 spaces though, and also the include files should be .inc.c instead of .c imo

File: src/C50A0.c [hunk 2]
```diff
@@ -184,7 +182,88 @@ s32 make_item_entity_delayed(s32 itemID, f32 x, f32 y, f32 z, s32 itemSpawnMode,
...
+    ItemEntity* item;
+    Shadow* shadow;
+    s32 i;
+    f32 posX;
+    f32 posY;
+    f32 posZ;
+    f32 depth;
+    s32 id;
+    
+    for(i = 0; i < MAX_ITEM_ENTITIES; i++) {
```
Comment (nanaian) [inline src/C50A0.c:196]:
```suggestion
    for (i = 0; i < MAX_ITEM_ENTITIES; i++) {
```

File: src/C50A0.c [hunk 3]
```diff
@@ -184,7 +182,88 @@ s32 make_item_entity_delayed(s32 itemID, f32 x, f32 y, f32 z, s32 itemSpawnMode,
...
+        if (gCurrentItemEntities[i] == NULL) {
+            break;
+        }
+    }
+    ASSERT(i < MAX_ITEM_ENTITIES);
+    id = i;
+    
+    gCurrentItemEntities[id] = item = heap_malloc(sizeof(ItemEntity));
```
Comment (nanaian) [inline src/C50A0.c:204]:
```suggestion
    gCurrentItemEntities[id] = item = heap_malloc(sizeof(*item));
```

File: src/item_entity_scripts.inc.c [hunk 4]
```diff
@@ -0,0 +1,1875 @@
+ItemScript ItemScript_Jump = {
```
Comment (nanaian) [inline src/item_entity_scripts.inc.c:1]: This should probably include the itemscript header

**Review by nanaian — CHANGES_REQUESTED** [review]

**Review by nanaian — APPROVED** [review]

**Pull request merged**